contentcuration/contentcuration/models.py

```diff
@@ -377,29 +378,43 @@ class ContentNode(MPTTModel, models.Model):
     objects = TreeManager()
 
     def get_original_node(self):
-
+        key = "original_channel_{}".format(self.original_source_node_id or self.pk)
+        cached_data = cache.get(key)
+        if cached_data:
+            return cached_data
+        original_node = self.original_node or self
         if self.original_channel_id and self.original_source_node_id:
             original_channel = Channel.objects.get(pk=self.original_channel_id)
-            return original_channel.main_tree.get_descendants().filter(node_id=self.original_source_node_id).first() or self
-
-        return self.original_node or self
+            original_node = original_channel.main_tree.get_descendants().filter(node_id=self.original_source_node_id).first() or self
+        cache.set(key, original_node, None)
+        return original_node
+
+    def get_associated_presets(self):
+        key = "associated_presets_{}".format(self.kind_id)
+        cached_data = cache.get(key)
+        if cached_data:
+            return cached_data
+        presets = FormatPreset.objects.filter(kind=self.kind).values()
+        cache.set(key, presets, None)
+        return presets
 
     def get_channel(self):
         root = self.get_root()
-        channel = root.channel_main or root.channel_trash or root.channel_language or root.channel_previous
+        channel = root.channel_main or root.channel_trash or root.channel_staging or root.channel_previous
         if channel:
             return channel.first()
```

Comment:
This was fixed elsewhere, right? Could we update this PR from master, resolving the merge conflicts?